### DISCUSSION TITLE
Api analytics slow

### DISCUSSION THREAD

**Discussion by alesiestu** [freekmurze/laravel-analytics#378] [Q&A]
Hi, 

i have a problem with analytics, for a few days the speed of the queries is very slow, I saw that there are similar reports on google, do you encounter the same problem? Also I often have google 502 errors:

`  </style>
  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
  <p><b>502.</b> <ins>That’s an error.</ins>
  <p>The server encountered a temporary error and could not complete your request.<p>Please try again in 30 seconds.  <ins>That’s all we know.</ins>

  at dashboard.relinko.net/vendor/google/apiclient/src/Http/REST.php:128
    124|       // if we errored out, it should be safe to grab the response body
    125|       $body = (string) $response->getBody();
    126| 
    127|       // Check if we received errors, and add those to the Exception for convenience
  > 128|       throw new GoogleServiceException($body, $code, null, self::getResponseErrors($body));
    129|     }
    130| 
    131|     // Ensure we only pull the entire body into memory if the request is not
    132|     // of media type

      [2m+11 vendor frames [22m
  12  dashboard.relinko.net/app/Helpers/CronHelper.php:39
      Spatie\Analytics\Analytics::performQuery(Object(Spatie\Analytics\Period), "ga:sessions", ["ga:sessions", "ga:medium,ga:source, ga:date", "ga:source==139", "-ga:sessions"])

  13  dashboard.relinko.net/app/Console/Commands/calcologiornaliero.php:95
      App\Helpers\CronHelper::getViewSearchCRON2(Object(Carbon\Carbon), Object(Carbon\Carbon))`

**Comment by mbardelmeijer** [comment]
We also have this issue also a few days now. I don't think it's due to this package, but more on the API side @ Google.

**Reply by alesiestu** [reply]
yes, i have open the iusse bug on this page:

https://issuetracker.google.com/issues?q=componentid:187400

please you too open a report, the more the better and google fix :D

**Comment by alesiestu** [comment]
this package is fantastic, i know that the problem is Google :)
